internal/sidecarlogresults/sidecarlogresults_test.go

```diff
@@ -0,0 +1,148 @@
+package sidecarlogresults
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/test/diff"
+)
+
+func TestLookForResults_FanOutAndWait(t *testing.T) {
+	for _, c := range []struct {
+		desc    string
+		results []SidecarLogResult
+	}{{
+		desc: "multiple results",
+		results: []SidecarLogResult{{
+			Name:  "foo",
+			Value: "bar",
+		}, {
+			Name:  "foo2",
+			Value: "bar2",
+		}},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			dir := t.TempDir()
+			resultNames := []string{}
+			wantResults := []string{}
+			for _, result := range c.results {
+				createResult(t, dir, result.Name, result.Value)
+				resultNames = append(resultNames, result.Name)
+				encodedResult, err := json.Marshal(result)
+				if err != nil {
+					t.Error(err)
+				}
+				// encode adds a newline character at the end.
+				// We need to do the same before comparing
+				encodedResult = append(encodedResult, '\n')
+				wantResults = append(wantResults, string(encodedResult))
+			}
+			dir2 := t.TempDir()
+			go createRun(t, dir2, false, true)
+			got := new(bytes.Buffer)
+			err := LookForResults(got, dir2, dir, strings.Join(resultNames, ","))
```

Comment:
If we're exporting this and think other clients will use this I'd lean towards taking in a []string rather than making clients need to encode as a csv.

Reply:
or if you want to keep the behavior of being able to pass in a single name you could use `...string`

Reply:
I agree, []string is much better. I had it as a csv because that's the input I was getting via cmd line flags. I moved the split logic into cmd/sidecarlogresults/main.go and pass in the []string to LookForResults.